packages/components/src/client/components/ArtPlayer.ts

```diff
@@ -0,0 +1,211 @@
+/* eslint-disable vue/no-unused-properties */
+import { PropType, defineComponent, h, onBeforeUnmount, onMounted } from "vue";
+import { deepMerge } from "vuepress-shared/client";
+import { useSize } from "../composables/index.js";
+import { autoType, flv, m3u8, mpd } from "../utils/index.js";
+
+import type { VNode } from "vue";
+
+import type {
+  ArtPlayer,
+  ArtPlayerOptions,
+  ComponentsArtPlayerOptions,
+  CustomArtPlayerOptions,
+} from "../../shared/index.js";
+
+declare const ART_PLAYER: ComponentsArtPlayerOptions;
+
+const optionsDefault = {
+  width: "100%",
+  ratio: 16 / 9,
+  src: {
+    fullscreen: true,
+    autoSize: true,
+    setting: true,
```

Comment:
Why not move these props to components. like `noFullscreen` `noAutosize` `noSetting`?

Reply:
These can be configured in props.src, and usually the same configuration of the website, I think I need to load the default settings on the node side, and then pass in the client

Reply:
Users may need component scope settings

Reply:
> Users may need component scope settings

OK, it can do like this

```html
<ArtPlayer :src="src" />

<script setup>
const src = {
  autoplay: false,
  fullscreen: false,
  // others
}
</script>
```
src will be passed to artplayer as the original parameter, I think you only need to put commonly used settings items in props.